### DISCUSSION TITLE
Why Hasn't This Become Standard?

### DISCUSSION THREAD

**Discussion by Wontfallo** [NVIDIA/Stable-Diffusion-WebUI-TensorRT#347] [General]
Hey, I'm really confused about why this isn't a top priority for Nvidia. It's been a year, and it only works with automatic1111 webui and not consistently.

When it does work, it's incredible! Imagine generating 1024x1024 SDXL images in just 2.3 seconds at 80 steps. It's mind-blowing. So, what's the deal, Nvidia? Why aren't the developers of ComfyUI, Forge, and Fooocus supporting this? It should be a top priority; it's such a waste that it's not. This could be a game-changer for Flux, and it would save so much energy since generation times would be reduced by seven times.

Seriously, can someone explain why this hasn't become the standard and why it's not supported anywhere else? Nvidia RTX graphics cards are the best for inference and image generation, but they're not even close to reaching their full potential. Do people just not know about this? If that's the case, we can educate them, but what's Nvidia's excuse for not supporting this more?

**Comment by T9es** [comment]
From what I understand:

1. There's no support for it since it's a half baked project. There are several issues that are simple to fix, but they are not getting fixed. Only the community is patching all the bugs. The other interfaces like forge UI are using some other methods that would need a fully other solution for implementation. Forge for instance doesn't have a very good documentation on all of this. There's a dev who made an alternate version of tensorRT (with additional features). When I asked about making it run on Forge, he responded after a few weeks that the docs on forge are not good enough to make this a viable thing. 

One other issue that was easy to fix was: spaces and dots in the loRA name would brick the exported engine. The fix was to modify a line or several, not even something complex, 5 minutes of work tops for someone that doesn't understand code. It was unfixed from the first version to the second one. 

2. Engines with SDXL take A LOT of space. Last I tested a month ago, loRA models are taking 7-8gb PER ENGINE. So 10 models would take 70-80gb. This is a lot of wasted space. There were talks about making the sizes really small, about a few MB at most, but it's radio silent. It's been 1.5 years since I last heard of this idea.

3. (could be involved with the N.O 2 point), with SDXL, loading the engine takes even more VRAM. On my 3090 OOM errors were really close to happening. 24GB of VRAM down the drain JUST loading all the models. 

Or it's likely NVIDIA dropped this project fully. 7 months without any update is a REALLY long time. Might be they are waiting for some breakthrough within torch or some other place. For now it's safe to say the project is dead.
